app/models/court.rb

```diff
@@ -7,4 +7,13 @@ class Court < ApplicationRecord
     concrete: 1,
     tartan: 2
   }
+
+  def available_at?(date)
+    return false if reservations.school_reservations.where("(EXTRACT(day FROM time_from) = ?
+      AND (EXTRACT(hour FROM time_from) < ? OR (EXTRACT(hour FROM time_from) = ? AND EXTRACT(min FROM time_from) < ?))
+      AND (EXTRACT(hour FROM time_to) > ? OR (EXTRACT(hour FROM time_to) = ? AND EXTRACT(min FROM time_to) > ?)))",
+      date.wday+2, date.hour, date.hour, date.min, date.hour, date.hour, date.min).present?
```

Comment:
Alternatively, you can use named placeholders in the template, and pass a hash as the second element of the array. The names in the template are replaced with the corresponding values from the hash.
```ruby
User.where(["name = :name and email = :email", { name: "Joe", email: "joe@example.com" }])
# SELECT * FROM users WHERE name = 'Joe' AND email = 'joe@example.com';
```

This can make for more readable code in complex queries.
To na pewno tutaj się przyda

Reply:
Argument date jest datą, tak ? Czemu nie można tego w podobny sposób zrobić co people reservations ?

Reply:
Ponieważ musimy pominąc dokłądną date i wziąć pod uwagę tylko godziny i dzień tygodnia.